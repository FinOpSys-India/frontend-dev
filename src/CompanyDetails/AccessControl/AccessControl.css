.accessNav{
    display: flex;
    width:100%;
    gap :419px;
    margin-top: -3px;

    span{
        font-family: Inter;
        font-size: 18px;
        font-weight: 600;
        line-height: 16px;
        text-align: left;
        padding-top: 15px;
        padding-left: 29px;  
        /* add this paddimg evberywhere */
    }


    .dropdown {
        padding: 8px 16px 8px 16px;
        gap: 8px;
        width:30%;
        

        .searchBarAccess{
            background-color: white;
            height :27px;
            padding-left :3px;
            margin-left: 12px;
            margin-right: 12px;
            border-radius: 12px;
        }

        button, .dropdown-item {
             border: 2px  solid gba(240, 240, 240, 1);
             background-color: rgba(240, 240, 240, 1);
             border-radius: 12px;
            font-size: 12px;
            font-weight: 500;
            /* line-height: 16px; */
            width: 100%;
            text-align: start;
            padding-left: 19px;
        }

         .dropdown-menu {
            background-color: rgba(240, 240, 240, 1);
            border-radius: 12px;
            font-size: 12px;
            font-weight: 500;
           /* line-height: 16px; */
            width: 89%;
         }
    }

}


/* ---------------------------boc  nav----------------------- */

.settingNameAndButtons{
    display: flex;
    width:100%;
    gap: 549px;
    margin-top: 26px;
    text-align: center;
    justify-content: center;
    justify-items: center;

    h5{
        font-size: 14px;
        font-weight: 500;
        line-height: 16px;
        padding-left: 3px;
        padding-top: 3px;
    }

    .toggleAndReset{
        display: flex;
        gap :12px;
        /* text-align: center;
        justify-content: center;
        justify-items: center; */

        .toggleDiv{
            background-color:  rgba(240, 240, 240, 1);
            padding-left: 6px;
            border-radius: 6px;

            .form-check-input:focus {
                border-color: none; /* or border-color: initial; */
                outline: none; /* or outline: initial; */
                box-shadow: none; /* or box-shadow: initial; */
            }
            
            

        }


      
           .resetDiv{
                background-color:  rgba(240, 240, 240, 1);
                padding-left: 2px;
                padding-right: 6px;
                padding-bottom: -30px;
                border-radius: 6px;
            }
    
       
        input{
               padding-right: 30px;
               width:60%
            
        }

        span{
            font-size: 12px;
            font-weight: 500;
            /* line-height: 16px; */
            /* padding-left: 24px; */
        }
    }
}

.accessControlDiv{
    display: flex;
    padding-left: 24px;
    flex-direction: column;
    gap: 12px;
    border-radius: 12px;
    opacity: 0px;
    background-color:  rgba(240, 240, 240, 1);
    margin-left: 27px;
    margin-top: 12px;
    padding-bottom: 30px;


    h6{
        font-size: 12px;
        font-weight: 500;
        margin-top: 9px;
    }

    .accessSpan{
            font-size: 10px;
            font-weight: 400;
            margin-top: 2px;
    }

    .accessControlInfo{
        display: flex;
        flex-direction: column;
        padding-left: 24px;
        width:  750px;
        margin-top: 12px;
        padding-bottom: 12px;
        gap: -1px;
        border-radius: 12px ;
        background-color: rgba(255, 255, 255, 1);
/* 
        span {
              margin-bottom: 12px;
        } */
    }


    .accessControlToggle{
        display: flex;
        flex-direction: row;
        gap:6px;
        position: relative;
        margin-top: -53px;
        margin-left: 610px;
        justify-content: center;
       width:12%;

        span{
             padding-left: 12px;
             padding-top: 3px;
        }
    }
}



.accessControlToggle{
    background-color:  rgba(240, 240, 240, 1);
    border-radius: 6px;
    width:15%;

    .toggleAndReset{
        /* margin-top: 106px; */
        /* padding-top: 12px; */
    }
    span{
        font-size: 12px;
        font-weight: 500;
    }
    .toggleDiv{
    
        .form-check-input:focus {
            border-color: none; /* or border-color: initial; */
            outline: none; /* or outline: initial; */
            box-shadow: none; /* or box-shadow: initial; */
        }
    }
} 
