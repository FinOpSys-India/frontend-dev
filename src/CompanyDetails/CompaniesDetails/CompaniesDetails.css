.companyDetails{
    margin-top: 15px;
}


.custom-file-upload {
    display: flex;
    flex-direction: column;
    color: #595959;
    font-weight: 400;
    font-size: 14px;
    width: 201px;
    line-height: 16px;
  }
  
  .add {
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    background-color: #7939ef;
    border-radius: 10px;
    gap: 8px;
    width: 141.8px;
    height: 28px;
    border: 0px;
    color: #ffffff;
    font-size: 12px;
    font-weight: 500px;
    line-height: 16px;
  }
  
  .addCompanyForm {
    padding: 34px;
    padding-top: 29px;
  }
  
  .company {
    width: 46px;
    height: 16px;
    color: #8c8c8c;
    font-weight: 500;
    font-size: 10px;
    opacity: 0.8;
    line-height: 16px;
    margin-bottom: 0px;
    margin-top: 30px;
  }
  
  .ds {
    width: 832px;
    height: 16px;
    font-weight: 600;
    color: #000000;
    font-size: 18px;
    line-height: 16px;
    margin-top: 0px;
    margin-bottom: 41px;
  }
  
  .dSetting {
    font-weight: 500;
    font-size: 10px;
    line-height: 16px;
    color: #8c8c8c;
    opacity: 0.8;
    height: 16px;
    margin-bottom: 0px;
  }
  
  .companyLogoSection {
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 20px;
  }
  
  .heading {
    color: #141414;
    font-size: 14px;
    line-height: 16px;
    font-weight: 500;
  }
  
  .firstCompany {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 770px;
    height: 110px;
    border-radius: 12px;
    padding: 24px;
    gap: 28px;
    background-color: #f0f0f0;
  }
  
  .firstBox {
    display: flex;
    flex-direction: column;
    margin-top: 6px;
  }
  


  .heading2 {
    color: #141414;
    font-size: 14px;
    line-height: 16px;
    font-weight: 500;
    margin-top: 36px;
  }
  

  .heading3 {
    color: #141414;
    font-size: 14px;
    line-height: 16px;
    font-weight: 500;
    margin-top: 36px;
  }
  
  .second {
    width: 770px;
    height: 190px;
  }
  
  .companyName {
    width: 91px;
    height: 15px;
    font-weight: 600;
    font-size: 120px;
    line-height: 14.52px;
    color: #000000;
  }
  
  .companyNameInput {
    width: 351px;
    height: 32px;
    border-radius: 10px;
    color: #000000;
    border: 0px;
    padding-left: 15px;
    font-size: 14px;
  }
  
  .marginLeft {
    margin-left: 30px;
  }
  
  .fontSize {
    font-size: 12px;
    font-weight: 600;
    color: #141414;
  }
  
  .endButtons {
    /* width: 876px; */
    display: flex;
    justify-content: end;
    gap: 10px;
    margin-top: 15px;
  }
  
  .save {
    background-color: #7939ef;
    padding: auto;
    border-radius: 10px;
    width: 60px;
    height: 28px;
    font-weight: 500;
    font-size: 12px;
    color: #ffffff;
    line-height: 16px;
    border: 0px;
    cursor: pointer;
  }
  
  .backToList {
    background-color: #7939ef;
    padding: auto;
    border-radius: 10px;
    width: 90px;
    height: 28px;
    font-weight: 500;
    font-size: 12px;
    color: #ffffff;
    line-height: 16px;
    border: 0px;
    cursor: pointer;
    margin-right: 29px;
  }
  
  .reset {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #f0f0f0;
    padding: auto;
    border-radius: 10px;
    width: 30px;
    cursor: pointer;
    height: 28px;
    color: #374957;
    border: 0px;
  }
  
  .sign {
    display: flex;
    align-items: center;
    gap: 8px;
  }
  
  .buttons {
    display: flex;
    gap: 8px;
  }
  
  .but-searchList{
    border: 0px;
    border-radius: 10px;
    background-color: #f0f0f0;
    width: 170px;
    height: 28px;
    padding-left: 60px;
    font-weight: 500;
    font-size: 12px;
    line-height: 16px;
    color: #141414;
  }
  
  .text {
    width: 463.2px;
    height: 16px;
    font-weight: 600;
    font-size: 16px;
    line-height: 16px;
    color: #000000;
  }

  .company-line {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 10px;
    padding-left: 20px;
    padding-right: 25px;
  }
  
  .companiesList {
    width: 100%;
    height: max-content;
    border: 0px;
    border-radius: 8px;
    padding: 24px;
    background-color: #f0f0f0;
  }
  
  .companyItem {
    display: flex;
    align-items: center;
    gap: 5px;
    /* width: 786px; */
    margin: 5px;
    height: 48px;
    box-shadow: 10px 10px 20px rgba(0, 0, 0, 0.1);
    border-radius: 10px;
    background-color: #ffffff;
    padding-left: 24px;
    margin-bottom: 12px;
  }
  
  .main {
    padding: 24px;
  }
  


  /* ---------------------------------------company list------------------------- */
  
  
  .cName {
    width: 676px;
    height: 16px;
    font-weight: 400;
    font-size: 12px;
    line-height: 16px;
    color: #141414;
    margin-bottom: -1px;
    margin-top: 15px;
  }
  
  .cEid {
    margin-top: 0px;
    color: #8c8c8c;
    font-weight: 400;
    font-size: 10px;
    line-height: 16px;
    width: 68px;
    height: 16px;
  }
  
  .arrow {
    display: flex;
    justify-content: end;
    align-items: center;
    margin-left: -56px;
  }
  
  .phoneDiv {
    display: flex;
    gap: 5px;
  }
  
  .enp {
    border: 0px;
    border-radius: 10px;
    width: 40px;
    height: 32px;
    padding-left: 15px;
  }
  
  input[type="number"]::-webkit-outer-spin-button,
  input[type="number"]::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  
  input[type="number"] {
    appearance: textfield;
    -webkit-appearance: none;
    -moz-appearance: textfield;
  }
  
  .companyNameInput:focus {
    outline: none;
    padding-left: 15px;
  }
  
  .companyName {
    width: 832px;
    height: 16px;
    font-weight: 600;
    font-size: 18px;
    line-height: 16px;
    color: #000000;
    margin-left: 33px;
  }
  
  .heading {
    width: 832px;
    height: 16px;
    font-weight: 500;
    font-size: 14px;
    line-height: 1px;
    color: #141414;
  }
  
  .logo {
    font-weight: 400px;
    font-size: 12px;
    line-height: 16px;
    color: #141414;
  }



  /* --------------------------------hovber------------------------- */


  .editable-field {
    position: relative;
    display: inline-block;
  }
  
  .editable-field input {
    border: none;
    background: transparent;
    padding: 5px;
    /* width: 100%; */
    box-sizing: border-box;
  }
  
  .editable-field .edit-icon {
    display: none;
    position: absolute;
    right: 10px;
    top: 50%;
    transform: translateY(-50%);
    cursor: pointer;
  }
  
  .editable-field input:focus {
    outline: none;
    border-bottom: 1px solid #000;
    background-color: white;
  }
  
  .editable-field:hover .edit-icon {
    display: block;
  }
  
  .editable-field:hover input {
    background-color: white;
  }
  
  


  /* ---------------------email------------ */
  #email{
    background-color: #f0f0f0;
  }