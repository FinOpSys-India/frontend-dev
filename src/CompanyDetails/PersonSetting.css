.dashboardPop{
  display: flex;
  flex-direction: row;
  margin-top: 0%;
  padding-top: 0%;
  /* position: absolute; */
  width:100%;
  height :100%;



  /* ---------------------------leftMenu------------------------ */

  .leftMenu{
    background-color: rgba(240, 240, 240, 1);
    /* position: absolute; */
    top: 0;
    left: 0;
    width:21%;
    Padding-left: 19px;
    Padding-top: 19px;
    Padding-left: 16px;
    margin-top: -16px;
    margin-left: -16px;

    .imageAndDetails{
      display: flex;
      flex-direction: row;
      width: 212px;
      height: 35px;
      gap: 14px;

    }

    .name{
      font-size: 14px;
      font-weight: 500;
    }

    .admin{
      font-size: 10px;
      font-weight: 500;
      color: rgba(188, 188, 188, 1);

    }


    .companySection, .dashboardSettings{
      h6{
        margin-top: 21px;
        font-size: 10px;
        font-weight: 500;
        line-height: 16px;
        /* text-align: left; */
        opacity: 0.8;
        color: rgba(140, 140, 140, 1);
      }


      button{
        padding: 8px 16px 8px 16px;
        width:90%;
        border-radius: 12px;
        opacity: 0px;
        font-size: 12px;
        text-align: start;
        padding-top: 6px;
        /* background-color: rgba(121, 57, 239, 1); */
        border: 0;
        font-weight: 500;
        margin-top: 6px;

      }
      
    }
  }






  /* --------------------------------rightMenu---------------------- */
  .rightSide{
    
    width:79%;
    /* margin-top: -16px; */
    /* margin-right: -120000px; */

    
  }

}


.avatarLogo{
  width: 33px;
  height: 36px;
  gap: 0px;
  opacity: 0px;
  border-radius: 11px 0px 0px 0px;
}
