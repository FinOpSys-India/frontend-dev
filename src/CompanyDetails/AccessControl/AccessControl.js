import React, { useEffect, useState } from 'react'
import RotateLeftIcon from '@mui/icons-material/RotateLeft';
import "./AccessControl.css"
import SearchIcon from '@mui/icons-material/Search';
import Demo from '../Demo/Demo';
import axios from "axios"

function AccessControl() {

    const [isOpen, setIsOpen] = useState(false);
    const [isChecked, setIsChecked] = useState(false);
    const [isSearchMode, setIsSearchMode] = useState(false); // State to track search mode
    const [isCheckedNotification, setIsCheckedNotification] = useState({
           generalAccess: "",
           notification:"",
           page1:""
        }
    );
    // const [isCheckedGeneralAccess, setIsCheckedGeneralAccess] = useState(false);

    const toggleDropdown = () => {
        setIsOpen(!isOpen);
        setIsChecked(false); // Reset isChecked state
        setIsSearchMode(!isSearchMode); // Toggle search mode
    };


    const toggleSwitch = () => {
        setIsChecked(!isChecked);
    };


useEffect(()=>{

    console.log(isCheckedNotification)

    axios.post('http://localhost:9000/update-notification', isCheckedNotification, { withCredentials: true })
    .then(response => {
        console.log(response.data);
        console.log(response.data.message);
        console.log(response.data.isCheckedNotification);
    })
    .catch(error => {
        console.error('Error sending notification status', error);
    });

},[isCheckedNotification])

  return (
    <div>

        <nav className='accessNav'>
            <span>Access Control</span>
            <div className='dropdown' >
                <button className="btn dropdown-toggle " type="button"   id="dropdownMenuButton1"  onClick={toggleDropdown}  aria-expanded={isOpen ? 'true' : 'false'}> Search Members</button>
                <ul className={`dropdown-menu ${isOpen ? 'show' : ''}`} aria-labelledby="dropdownMenuButton1">
                   <div className='searchBarAccess'> {}<SearchIcon style={{fontSize: 16,  marginTop:"6px" , marginLeft:"9px"}}/></div>
                    {

                        <li><a className="dropdown-item" href="#">Action</a></li>
                    }
                </ul>
            </div>
        </nav>

        <div className='accessBox'>
               <div className='settingNameAndButtons'>
                  <h5 >General Access</h5>
                  <div className='toggleAndReset '>
                       <div className='toggleDiv'>
                            <div class="form-check form-switch toggle" >
                                    <input class="form-check-input" data-testid="general-access-switch" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "15px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }}  checked={isCheckedNotification.generalAccess}  onChange={(e) =>  setIsCheckedNotification({ ...isCheckedNotification, generalAccess: !isCheckedNotification.generalAccess})  } />
                            </div>
                       </div>

                        <div className='resetDiv'>
                            <span> <RotateLeftIcon style={{ fontSize: 16}}/> Reset1</span>
                        </div>
                  </div>
               </div>


               <div className='mainDiv'>
                   
                    <div className='accessControlDiv'>
                        <div className='accessControlInfo'>
                            <h6>Notification </h6>
                            <span className='accessSpan'>We need to write two line short description about a tool.</span>
                        </div>

                        <div className='accessControlToggle toggleAndReset'>
                               <span> {isCheckedNotification.notification  ? "Active" : "Inactive"} </span>
                               <div className='toggleDiv'>
                                    <div class="form-check form-switch toggle" > 
                                            <input class="form-check-input"  data-testid="notification-switch" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "30px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }} checked={isCheckedNotification.notification}  onChange={(e) =>  setIsCheckedNotification({ ...isCheckedNotification, notification: !isCheckedNotification.notification })  } />
                                    </div>
                               </div>
                             {/* <div style={{display:"none"}}><Demo isCheckedNotification={isCheckedNotification}  onChange={(e) =>  setIsCheckedNotification({ ...isCheckedNotification, notification: e.target.value })  }/> </div>  */}
                        </div>



                        <div className='accessControlInfo'>
                            <h6>Page 1 </h6>
                            <span className='accessSpan'>We need to write two line short description about a tool.</span>
                        </div>

                        <div className='accessControlToggle toggleAndReset'>
                               <span>{isCheckedNotification.page1  ? "Active" : "Inactive"} </span>
                               <div className='toggleDiv'>
                                    <div class="form-check form-switch toggle" > 
                                            <input class="form-check-input" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "30px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }}  checked={isCheckedNotification.page1}  onChange={(e) =>  setIsCheckedNotification({ ...isCheckedNotification, page1: !isCheckedNotification.page1 })  }/>
                                    </div>
                               </div>
                        </div>
                       


                        <div className='accessControlInfo'>
                            <h6>Page 2</h6>
                            <span className='accessSpan'>We need to write two line short description about a tool.</span>
                        </div>

                        <div className='accessControlToggle toggleAndReset'>
                               <span>Active</span>
                               <div className='toggleDiv'>
                                    <div class="form-check form-switch toggle" > 
                                            <input class="form-check-input" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "30px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }} checked={isChecked} onChange={toggleSwitch}/>
                                    </div>
                               </div>
                        </div>


                        <div className='accessControlInfo'>
                            <h6>Page 4</h6>
                            <span className='accessSpan'>We need to write two line short description about a tool.</span>
                        </div>

                        <div className='accessControlToggle toggleAndReset'>
                               <span>Active</span>
                               <div className='toggleDiv'>
                                    <div class="form-check form-switch toggle" > 
                                            <input class="form-check-input" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "30px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }} checked={isChecked} onChange={toggleSwitch}/>
                                    </div>
                               </div>
                        </div>

                     </div>

                </div>
        </div>





        <div className='accessBox'>
               <div className='settingNameAndButtons'>
                  <h5 >Account Payable</h5>
                  <div className='toggleAndReset '>
                       <div className='toggleDiv'>
                            <div class="form-check form-switch toggle" >
                                    <input class="form-check-input" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "15px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }} checked={isChecked} onChange={toggleSwitch}/>
                            </div>
                       </div>

                        <div className='resetDiv'>
                            <span> <RotateLeftIcon style={{ fontSize: 16}}/> Reset</span>
                        </div>
                  </div>
               </div>


               <div className='mainDiv'>
                   
                    <div className='accessControlDiv'>
                        <div className='accessControlInfo'>
                            <h6>Page </h6>
                            <span className='accessSpan'>We need to write two line short description about a tool.</span>
                        </div>

                        <div className='accessControlToggle toggleAndReset'>
                               <span>Active</span>
                               <div className='toggleDiv'>
                                    <div class="form-check form-switch toggle" > 
                                            <input class="form-check-input" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "30px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }} checked={isChecked} onChange={toggleSwitch}/>
                                    </div>
                               </div>
                        </div>




                        <div className='accessControlInfo'>
                            <h6>Page </h6>
                            <span className='accessSpan'>We need to write two line short description about a tool.</span>
                        </div>

                        <div className='accessControlToggle toggleAndReset'>
                               <span>Active</span>
                               <div className='toggleDiv'>
                                    <div class="form-check form-switch toggle" > 
                                            <input class="form-check-input" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "30px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }} checked={isChecked} onChange={toggleSwitch}/>
                                    </div>
                               </div>
                        </div>



                        <div className='accessControlInfo'>
                            <h6>Page </h6>
                            <span className='accessSpan'>We need to write two line short description about a tool.</span>
                        </div>

                        <div className='accessControlToggle toggleAndReset'>
                               <span>Active</span>
                               <div className='toggleDiv'>
                                    <div class="form-check form-switch toggle" > 
                                            <input class="form-check-input" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "30px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }} checked={isChecked} onChange={toggleSwitch}/>
                                    </div>
                               </div>
                        </div>


                        <div className='accessControlInfo'>
                            <h6>Page </h6>
                            <span className='accessSpan'>We need to write two line short description about a tool.</span>
                        </div>

                        <div className='accessControlToggle toggleAndReset'>
                               <span>Active</span>
                               <div className='toggleDiv'>
                                    <div class="form-check form-switch toggle" > 
                                            <input class="form-check-input" type="checkbox" id="flexSwitchCheckDefault" style={{ width: "30px", height: "15px", backgroundColor: isChecked ? 'rgba(22, 163, 74, 1)' : '' }} checked={isChecked} onChange={toggleSwitch}/>
                                    </div>
                               </div>
                        </div>

                     </div>

                </div>
        </div>




    </div>
  )
}

export default AccessControl

