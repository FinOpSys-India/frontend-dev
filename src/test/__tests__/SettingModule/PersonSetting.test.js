import { render, screen, fireEvent } from '@testing-library/react';
import PersonSetting from '../../../CompanyDetails/PersonSetting';
import AccessControl from '../../../CompanyDetails/AccessControl/AccessControl';
import Profile from '../../../CompanyDetails/Profile/Profile';
import TeamAndMember from '../../../CompanyDetails/TeamAndMember/TeamAndMember';
import CompaniesDetails from '../../../CompanyDetails/CompaniesDetails/CompaniesDetails';
import Integration from '../../../CompanyDetails/Integration/Integration';



// jest.mock('AccessControl');
// jest.mock('Profile');
// // jest.mock('../../../CompanyDetails/TeamAndMember/TeamAndMember', () => () => <div>TeamAndMember Component</div>);
// jest.mock('CompaniesDetails');
// jest.mock('../../../CompanyDetails/Integration/Integration', () => () => <div>Integration Component</div>);

describe('PersonSetting Component', () => {


  test('renders the component', async () => {
    render(<PersonSetting/>);
    expect(screen.getByText('Dashboard Settings')).toBeInTheDocument();
  });



  test('renders the correct component when buttons are clicked', async () => {
    render(<PersonSetting />);

    // Click the Profile button
    fireEvent.click(screen.getByText(/Profiles/i));
    expect(await screen.findByText('Profiles')).toBeInTheDocument();

    // Click the Companies button
    fireEvent.click(screen.getByText(/Company/i));
    expect(await screen.findByText('Company')).toBeInTheDocument();

    // // Click the Access Control button
    // fireEvent.click(screen.getByRole('button', { name: /Access Control/i }));
    // expect(await screen.findByText('Members')).toBeInTheDocument();
    
    
    // Click the Access Control button
     fireEvent.click(screen.getByRole('button', { name: /Team & Members/i }));
     expect(await screen.findByText('Team & Members')).toBeInTheDocument();



    // Click the Access Control button
    fireEvent.click(screen.getByRole('button', { name: /Notifications/i }));
    expect(await screen.findByText('Notifications')).toBeInTheDocument();

    // Click the Access Control button
    fireEvent.click(screen.getByRole('button', { name: /Security/i }));
    expect(await screen.findByText('Security')).toBeInTheDocument();

    // Click the Access Control button
    fireEvent.click(screen.getByRole('button', { name: /Integration/i }));
    expect(await screen.findByText('Connection')).toBeInTheDocument();

});



  test('highlights the active button', async () => {
    render(<PersonSetting />);

    // Click the Profile button
    const profileButton = screen.getByText(/Profiles/i);
    fireEvent.click(profileButton);
    expect(profileButton).toHaveClass('clickedButton');

    // Click the Companies button
    const companiesButton = screen.getByText(/Companies/i);
    fireEvent.click(companiesButton);
    expect(companiesButton).toHaveClass('clickedButton');
    expect(profileButton).not.toHaveClass('clickedButton');
  });




  test('matches the snapshot', async () => {
    const { asFragment } = render(<PersonSetting />);
    expect(asFragment()).toMatchSnapshot();
  });
});
