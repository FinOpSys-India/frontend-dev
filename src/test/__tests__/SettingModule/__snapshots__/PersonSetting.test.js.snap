// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonSetting Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="dashboardPop"
  >
    <div
      class="leftMenu"
    >
      <div
        class="imageAndDetails"
      >
        <img
          src="https://img.freepik.com/premium-vector/default-avatar-profile-icon-social-media-user-image-gray-avatar-icon-blank-profile-silhouette-vector-illustration_561158-3408.jpg"
        />
        <div>
          <h5
            class="name "
          >
            James Young
          </h5>
          <p
            class="admin "
          >
            Admin
          </p>
        </div>
      </div>
      <div
        class="companySection"
      >
        <h6>
          Company
        </h6>
        <button
          class=""
        >
           
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleOutlineIcon"
            focusable="false"
            style="font-size: 14px; margin-right: 9px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
            />
          </svg>
              Profiles 
        </button>
         
        <br />
        <button
          class=""
        >
           
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleOutlineIcon"
            focusable="false"
            style="font-size: 14px; margin-right: 9px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
            />
          </svg>
          Companies
        </button>
        <br />
        <button
          class=""
        >
           
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleOutlineIcon"
            focusable="false"
            style="font-size: 14px; margin-right: 9px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
            />
          </svg>
          Team & Members
        </button>
        <br />
        <button
          class=""
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleOutlineIcon"
            focusable="false"
            style="font-size: 14px; margin-right: 9px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
            />
          </svg>
          Access Control
        </button>
        <br />
      </div>
      <div
        class="dashboardSettings"
      >
        <h6>
          Dashboard Settings
        </h6>
        <button
          class=""
        >
           
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleOutlineIcon"
            focusable="false"
            style="font-size: 14px; margin-right: 9px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
            />
          </svg>
            Notifications 
        </button>
         
        <br />
        <button
          class=""
        >
           
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleOutlineIcon"
            focusable="false"
            style="font-size: 14px; margin-right: 9px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
            />
          </svg>
          Security
        </button>
        <br />
        <button
          class=""
        >
           
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleOutlineIcon"
            focusable="false"
            style="font-size: 14px; margin-right: 9px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
            />
          </svg>
          Integration
        </button>
        <br />
        <button
          class=""
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleOutlineIcon"
            focusable="false"
            style="font-size: 14px; margin-right: 9px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
            />
          </svg>
           Connection
        </button>
        <br />
      </div>
    </div>
    <div
      class="rightSide"
    />
  </div>
</DocumentFragment>
`;
